app/src/main/java/com/example/playlistmaker/SettingsActivity.kt

```diff
@@ -18,10 +18,16 @@ class SettingsActivity : AppCompatActivity() {
         val view = binding.root
         setContentView(view)
 
+        binding.switchDarkTheme.isChecked = (applicationContext as App).darkTheme
+
         binding.backButton.setOnClickListener{
             finish()
         }
 
+        binding.switchDarkTheme.setOnCheckedChangeListener { _, isChecked ->
+            (applicationContext as App).switchTheme(isChecked)
```

Comment:
```suggestion
            (application as App).switchTheme(isChecked)
```
Лучше использовать `application` в активити, если хочешь получить доступ к `App`, т.к. за контекстом может быть спрятан какой-нибудь ContextWrapper